framework/packages/abstract-client/src/account.rs

```diff
@@ -160,18 +160,73 @@ impl<Chain: CwEnv> Account<Chain> {
     // Install an application on the account
     // creates a new sub-account and installs the application on it.
     pub fn install_app<
-        M: ContractInstance<Chain> + RegisteredModule + From<Contract<Chain>> + Clone,
+        M: ContractInstance<Chain> + InstallConfig + From<Contract<Chain>> + Clone,
     >(
         &self,
         configuration: &M::InitMsg,
         funds: &[Coin],
+    ) -> AbstractClientResult<Application<Chain, M>> {
+        self.install_app_internal(vec![M::install_config(configuration)?], funds)
+    }
+
+    pub fn install_app_with_dependencies<
+        M: ContractInstance<Chain>
+            + DependencyCreation
+            + InstallConfig
+            + From<Contract<Chain>>
+            + Clone,
+    >(
+        &self,
+        module_configuration: &M::InitMsg,
+        dependencies_config: M::DependenciesConfig,
+        funds: &[Coin],
+    ) -> AbstractClientResult<Application<Chain, M>> {
+        let mut install_configs: Vec<ModuleInstallConfig> =
+            M::dependency_install_configs(dependencies_config)?;
+        install_configs.push(M::install_config(module_configuration)?);
+
+        self.install_app_internal(install_configs, funds)
+    }
+
+    pub fn admin(&self) -> AbstractClientResult<Addr> {
+        self.abstr_account.manager.address().map_err(Into::into)
+    }
+
+    pub fn module_infos(
```

Comment:
start_after and limit are really constraints that exist on the smart-contract level. As a developer they are a pain to work with so it'd be great if we could just get all the information and do the paging in this function. 

It'll be a recurring operation so could be worth to write a small macro to reduce the boilerplate for paging the queries.

Reply:
Good idea! I'll implement the macro later if necessary since for now we have just a single instance of this.